src/pkg/task/dao/execution.go

```
@@ -376,3 +402,77 @@ func buildInClauseSQLForExtraAttrs(keys []string) string {
 		return fmt.Sprintf("select id from execution where extra_attrs->%s->>?=?", s)
 	}
 }
+
+func buildExecStatusOutdateKey(id int64) string {
+	return fmt.Sprintf("execution:id:%d:status_outdate", id)
+}
+
+func extractExecIDFromKey(key string) (int64, error) {
+	key = strings.TrimPrefix(key, "execution:id:")
+	key = strings.TrimSuffix(key, ":status_outdate")
+	id, err := strconv.ParseInt(key, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	return id, nil
+}
+
+func (e *executionDAO) AsyncRefreshStatus(ctx context.Context, id int64) (err error) {
+	key := buildExecStatusOutdateKey(id)
+	if cache.Default().Contains(ctx, key) {
+		// return earlier if already have the key
+		return nil
+	}
+	// save the key to redis, the value is useless so set it to empty
+	return cache.Default().Save(ctx, key, "")
+}
+
+// scanAndRefreshOutdateStatus scans the outdate execution status from redis and then refresh the status to db,
+// do not want to expose to external use so keep it as private method.
+func (e *executionDAO) scanAndRefreshOutdateStatus(ctx context.Context) {
```

this method should return an error, and log error in the caller side.